frontend_tests/node_tests/emoji.js

```diff
@@ -78,15 +78,43 @@ zrequire('util');
 }());
 
 (function test_translate_emoticons_to_names() {
-  global.emoji_codes = {
-      codepoint_to_name: {
-          '1f603': 'smiley',
-      },
-  };
+    // Simple test
+    var test_text = 'Testing :)';
+    var expected = 'Testing :smiley:';
+    var result = emoji.translate_emoticons_to_names(test_text);
+    assert.equal(expected, result);
 
-  var test_text = 'Testing :)';
-  var expected = 'Testing :smiley:';
-  var result = emoji.translate_emoticons_to_names(test_text);
-
-  assert.equal(expected, result);
+    // Extensive tests.
+    // The following code loops over the test cases and each emoticon conversion
+    // to generate multiple test cases.
+    var EMOTICON_CONVERSIONS = {
+        ':)': 'smiley',
+        '(:': 'smiley',
+        ':(': 'slightly_frowning_face',
+        '<3': 'heart',
+        ':|': 'expressionless',
+        ':/': 'confused',
+    };
+    var testcases = [
+        {name: 'only emoticon', original: ':original:', expected: ':converted:'},
+        {name: 'space at start', original: ' :original:', expected: ' :converted:'},
+        {name: 'space at end', original: ':original: ', expected: ':converted: '},
+        {name: 'symbol at end', original: ':original:!', expected: ':converted:!'},
+        {name: 'symbol at start', original: 'Hello :original:', expected: 'Hello :converted:'},
+        {name: 'after a word', original: 'Hello:original:', expected: 'Hello:original:'},
+        {name: 'between words', original: 'Hello:original:World', expected: 'Hello:original:World'},
+        {name: 'end of sentence', original: 'End of sentence. :original:', expected: 'End of sentence. :converted:'},
+        {name: 'between symbols', original: 'Hello.:original:! World.', expected: 'Hello.:converted:! World.'},
```

Comment:
I think probably this test case should have the opposite result: we don't want non-symbols just before the symbol...

But probably obscure enough to be OK.

Reply:
Explain this one? Or, mark these cases as valid/invalid: 

- [ ] `hello.:)!`
- [ ] `hello:)!`
- [ ] `hello. :)!`
- [ ] `hello .:)!`

Do you want to have space before the emoticon? Or do you want a space before the symbol before the emoticon?

Reply:
Pushing an additional commit for this part, that you should be able to merge/not merge atomically. I'm making it such that if both sides of the text have non space symbols, then we do not do the conversion.